### PR TITLE
[Metricbeat] Migrate Apache status Metricset to use ReporterV2 interface

### PR DESCRIPTION
**Pull request by sayden**
Refer to elastic/beats#10774 for more info

### CODE REVIEW

**Comment by sayden** [comment]
jenkins, test this

**Comment by sayden** [comment]
Error in Elasticsearch module seems unrelated https://beats-ci.elastic.co/job/elastic+beats+pull-request+multijob-linux/beat=metricbeat,label=linux-immutable/5506/console

File: metricbeat/module/apache/status/status_test.go [hunk 1]
```diff
@@ -15,8 +15,6 @@
 // specific language governing permissions and limitations
 // under the License.
 
-// +build !integration
```
Comment (ruflin) [inline metricbeat/module/apache/status/status_test.go]: Why was this removed?

File: metricbeat/tests/system/test_apache.py [hunk 2]
```diff
@@ -55,7 +55,7 @@ def test_output(self):
             time.sleep(0.5)
 
         proc = self.start_beat()
-        self.wait_until(lambda: self.output_lines() > 0)
+        self.wait_until(lambda: self.output_lines() > 0, max_timeout=20)
```
Comment (ruflin) [inline metricbeat/tests/system/test_apache.py]: apache has a timeout of more then 10s?

**Comment by ruflin** [comment]
Same as other PR's, set it to `in progress`.

File: metricbeat/module/apache/status/status_test.go (hunk 1 above)
Comment (sayden) [reply metricbeat/module/apache/status/status_test.go]: Sorry, I usually have to remove it so Goland is able to work on that file. I forgot to undo it. Thanks!

File: metricbeat/tests/system/test_apache.py (hunk 2 above)
Comment (sayden) [reply metricbeat/tests/system/test_apache.py]: I can't remember why I changed this, frakly. I know there was a good reason :confused: Default is 10 seconds and I raised to 20 because I probably received some timeout in some moment. If you want I can leave it in 10 again until I remember :sweat_smile:

**Comment by sayden** [comment]
@ruflin I tried to use the new testing framework here but I was getting a 404. I'm not sure about why so I uploaded everything I had progressed.

**Comment by ruflin** [comment]
@sayden Will have a look.

File: metricbeat/module/apache/status/_meta/testdata/config.yml [hunk 3]
```diff
@@ -0,0 +1,3 @@
+type: http
+url: "/server-status"
```
Comment (ruflin) [inline metricbeat/module/apache/status/_meta/testdata/config.yml]: This needs to be `url: "/server-status?auto="` as it uses a param.

File: metricbeat/module/apache/status/_meta/testdata/input.plain [hunk 4]
```diff
@@ -0,0 +1,354 @@
+<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 3.2 Final//EN">
```
Comment (ruflin) [inline metricbeat/module/apache/status/_meta/testdata/input.plain]: This should look identical to https://github.com/elastic/beats/blob/master/metricbeat/module/apache/status/_meta/test/status_apache-2.4.23-CentOS6 and the others. You missed probably the `?auto` part in your query.

File: metricbeat/tests/system/test_apache.py (hunk 2 above)
Comment (ruflin) [reply metricbeat/tests/system/test_apache.py]: I prefer to go back to 10s as long as we don't know why.

File: metricbeat/module/apache/status/status_integration_test.go [hunk 5]
```diff
@@ -32,27 +32,25 @@ import (
...
 	t.Logf("%s/%s event: %+v", f.Module().Name(), f.Name(), event)
 
 	// Check number of fields.
-	if len(event) < 11 {
+	if len(event.MetricSetFields) < 11 {
 		t.Fatal("Too few top-level elements in the event")
 	}
 }
 
 func TestData(t *testing.T) {
```
Comment (ruflin) [inline metricbeat/module/apache/status/status_integration_test.go]: We can remove this one, right?
